packages/frontend/core/src/components/affine/share-page-modal/share-menu/share-page.tsx

```diff
@@ -153,6 +157,7 @@ export const AFFiNESharePage = (props: ShareMenuProps) => {
 
   const isMac = environment.isMacOs;
 
+  const { blockIds, elementIds } = useSelectedBlocks(currentMode);
```

Comment:
`useSelectedBlocks` It makes me feel like we are subscribe selected blocks, but in fact we only read once, what about we write useMemo here?
```suggestion
  const { blockIds, elementIds } = useMemo(
    () => getSelectionIds(editor?.host || null, mode),
    [editor, mode]
  );
```